### PR TITLE
Refactor artefact metadata routes and rendering

### PR DESCRIPTION
**Pull request by 8R0WNI3**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

**Review by zkdev — CHANGES_REQUESTED** [review]

File: src/api.js [hunk 1]
```diff
@@ -313,34 +313,37 @@ const componentsComplianceSummary = async ({
 
 
 const artefactsQueryMetadata = async ({
-  components,
+  artefacts,
   types,
   referenced_types,
+  enableCache,
```
[the remark this answers is not in the document]
Comment (8R0WNI3) [reply src/api.js:319]: Agree 👍🏼

File: src/components/dependencies/ComplianceCells.js [hunk 2]
```diff
@@ -55,94 +55,94 @@ import CloudDownloadIcon from '@mui/icons-material/CloudDownload'
...
+            size='small'
+          /> : <Chip
+            label={`${osInfo.ID} ${osInfo.VERSION_ID}`}
+            color={severity.color}
+            variant='outlined'
+            size='small'
+          />
+        ) : (
+          isLoading ? <Chip
+            label='OS Info Loading...'
```
Comment (zkdev) [inline src/components/dependencies/ComplianceCells.js:125]: rather use circularprogress or skeleton

File: src/components/dependencies/ComplianceCells.js [hunk 3]
```diff
@@ -212,64 +212,65 @@ const findLastScan = (complianceData, datasource) => {
 }
 
 
+const lastScanTimestampStr = (lastScan, isLoading) => {
+  if (isLoading) return 'Loading...'
```
Comment (zkdev) [inline src/components/dependencies/ComplianceCells.js:216]: also here, rather use a loading animation than static text

File: src/components/dependencies/ComplianceCells.js [hunk 4]
```diff
@@ -294,38 +295,38 @@ const ComplianceCell = ({
...
           scanConfig={singleScanCfgOrNull({scanCfgs: scanConfigs, complianceToolName: COMPLIANCE_TOOLS.BDBA})}
-          fetchComplianceData={fetchComplianceData}
           fetchComplianceSummary={fetchComplianceSummary}
+          isLoading={isDataLoading}
+        />
+      }
+      {
+        artefact.kind === ARTEFACT_KIND.RESOURCE && <GolangChip
+          versions={structureInfos?.filter((structureInfo) => {
+            return structureInfo.data.package_name === 'golang-runtime'
```
Comment (zkdev) [inline src/components/dependencies/ComplianceCells.js:308]: factor out golang package name to const, please

**Review by zkdev — CHANGES_REQUESTED** [review]

**Pull request merged**